aas-gui/Frontend/aas-web-gui/src/mixins/SubmodelElementHandling.ts

```diff
@@ -463,6 +471,24 @@ export default defineComponent({
                 path = selectedNode.path + '/attachment';
             }
             return path;
-        }
+        },
+
+        // Get the Unit from the EmbeddedDataSpecification of the ConceptDescription of the Property (if available)
```

Comment:
I gave this some thought and discussed this with colleagues. I think you are right. We should use for instead of foreach for the sake of potential performance improvements and code clarity even tho we use foreachs everywhere else. Here, a for-loop would make more sense. Please update the method to this:

```js
// Get the Unit from the EmbeddedDataSpecification of the ConceptDescription of the Property (if available)
        unitSuffix(prop: any) {
            if (!prop.conceptDescriptions) {
                return '';
            }
            for (const conceptDescription of prop.conceptDescriptions) {
                if (!conceptDescription.embeddedDataSpecifications) {
                    continue;
                }
                for (const embeddedDataSpecification of conceptDescription.embeddedDataSpecifications) {
                    if (embeddedDataSpecification.dataSpecificationContent && embeddedDataSpecification.dataSpecificationContent.unit) {
                        return embeddedDataSpecification.dataSpecificationContent.unit;
                    }
                }
            }
            return '';
        },
```